crates/llvm-context/src/polkavm/evm/ext_code.rs

```diff
@@ -45,11 +45,30 @@ where
 
 /// Translates the `extcodehash` instruction.
 pub fn hash<'ctx, D>(
-    _context: &mut Context<'ctx, D>,
-    _address: inkwell::values::IntValue<'ctx>,
+    context: &mut Context<'ctx, D>,
+    address: inkwell::values::IntValue<'ctx>,
 ) -> anyhow::Result<inkwell::values::BasicValueEnum<'ctx>>
 where
     D: Dependency + Clone,
 {
-    todo!()
+    let address_type = context.integer_type(revive_common::BIT_LENGTH_ETH_ADDRESS);
+    let address_pointer = context.build_alloca_at_entry(address_type, "address_pointer");
+    let address_truncated =
+        context
+            .builder()
+            .build_int_truncate(address, address_type, "address_truncated")?;
+    let address_swapped = context.build_byte_swap(address_truncated.into())?;
+    context.build_store(address_pointer, address_swapped)?;
+
+    let extcodehash_pointer = context.build_alloca(context.word_type(), "extcodehash_pointer");
+
+    context.build_runtime_call(
```

Comment:
This does not fully resemble the [EVM semantics](hash: hash of the chosen account's [code](https://www.evm.codes/about), the empty hash (0xc5d24601...) if the account has no [code](https://www.evm.codes/about), or 0 if the account does not exist or has been destroyed.). Here is [the syscall](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/revive/src/wasm/runtime.rs#L1409) for reference.

I think with the current implementation in `pallet-revive`, we are basically forced to emit some branches and make two syscalls for this. Which is not desired, everything the runtime can do for us easily it should do (it is faster and spares contract code size). So this needs a PR in the pallet first to let it distinguish between contract-accounts, no-contract accounts and unknown accounts and write back the correct value automatically. Then it can be properly implemented and tested.

Sorry in the corresponding issue #63 I should have wrote that out more clearly.